### PR TITLE
Feat/expose parc merge

### LINKED ISSUE #146: [ENH]: Obtain merged parcellation as niimg-like object or nifti file
### Are you requiring a new dataset or marker?

- [X] I understand this is not a marker or dataset request

### Which feature do you want to include?

If one extracts features using merged parcellations, it is still often useful or necessary to have a copy of the parcellation used to extract features for follow-up analysis. Junifer should have some easy-to-use function to provide exactly that.

### How do you imagine this integrated in junifer?

Put it in a function. Its already more or less implemented in ParcelAggregation.

### Do you have a sample code that implements this outside of junifer?

```shell
Put it in a function. Its already more or less implemented in ParcelAggregation.
```

### Anything else to say?

_No response_

### PR DESCRIPTION
**Pull request by LeSasse**
* [x] fix #146
* [x] description of feature/fix
* [x] tests added/passed
* [x] add an entry to the [latest changes](../docs/changes/latest.inc)

This exposes the merging of parcellations (for example cortical and subcortical parcellations, i.e. schaefer + tian parcellations) as a function for the user, so that they can have merged parcellations available for subsequent analyses.

### CODE REVIEW

**Review by fraimondo — CHANGES_REQUESTED** [review]
@LeSasse :

Can you copy this tests and adapt them to the `merge_parcellations` test?

https://github.com/juaml/junifer/blob/af2a8cf999949624859594d7f8522035ff02ac78/junifer/markers/tests/test_parcel_aggregation.py#L293

They are basically testing overlapping / duplicated labels and stuff like that. This should be also tested here as it is no longer the responsibility of the marker to deal with several parcellations.

**Comment by fraimondo** [comment]
Rest is perfect, just need the proper tests.

File: junifer/data/parcellations.py [hunk 1]
```diff
@@ -700,3 +700,84 @@ def _retrieve_suit(
     ].to_list()
 
     return parcellation_fname, labels
+
+
+def merge_parcellations(
+    parcellations_list: List,
```
Comment (synchon) [inline junifer/data/parcellations.py:706]: Is it `List[str]`?

File: junifer/data/parcellations.py [hunk 2]
```diff
@@ -700,3 +700,84 @@ def _retrieve_suit(
...
+    parcellations_names: List,
+    labels_lists: List[List],
+) -> Tuple:
+    """Merge all parcellations from a list into one parcellation.
+
+    Parameters
+    ----------
+    parcellations_list : list
```
Comment (synchon) [inline junifer/data/parcellations.py:714]: The basic types like str can be documented here as well for ease.

**Review by synchon — CHANGES_REQUESTED** [review]

File: docs/changes/latest.inc [hunk 3]
```diff
@@ -63,6 +63,8 @@ Enhancements
 
 - Improve metadata and data I/O for :class:`junifer.storage.HDF5FeatureStorage` (:gh:`196` by `Synchon Mandal`_).
 
+- Expose a function to merge a list of parcellations (:gh:`146` by Leonard Sasse`_).
```
[the remark this answers is not in the document]
Comment (LeSasse) [reply docs/changes/latest.inc:66]: > @LeSasse :
> 
> Can you copy this tests and adapt them to the `merge_parcellations` test?
> 
> https://github.com/juaml/junifer/blob/af2a8cf999949624859594d7f8522035ff02ac78/junifer/markers/tests/test_parcel_aggregation.py#L293
> 
> They are basically testing overlapping / duplicated labels and stuff like that. This should be also tested here as it is no longer the responsibility of the marker to deal with several parcellations.

ok, will check it out!

File: junifer/data/parcellations.py (hunk 1 above)
Comment (LeSasse) [reply junifer/data/parcellations.py:706]: no, it should be `List["Nifti1Image"]` i think, should i put it?

File: junifer/data/parcellations.py [hunk 4]
```diff
@@ -700,3 +700,84 @@ def _retrieve_suit(
     ].to_list()
 
     return parcellation_fname, labels
+
+
+def merge_parcellations(
+    parcellations_list: List,
+    parcellations_names: List,
+    labels_lists: List[List],
+) -> Tuple:
```
[the remark this answers is not in the document]
Comment (LeSasse) [reply junifer/data/parcellations.py:709]: makes sense, for the Tuple it works as `Tuple["Nifti1Image", List[str]]`?

File: junifer/data/parcellations.py (hunk 1 above)
Comment (synchon) [reply junifer/data/parcellations.py:706]: Yeah

File: junifer/data/parcellations.py (hunk 2 above)
Comment (LeSasse) [reply junifer/data/parcellations.py:714]: you mean `parcellations_list : list of Nifti1Image`?

File: junifer/data/parcellations.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (synchon) [reply junifer/data/parcellations.py:709]: Yeah exactly.

File: junifer/data/parcellations.py (hunk 2 above)
Comment (synchon) [reply junifer/data/parcellations.py:714]: Yeah but I think it should be `... niimg-like object` as `nibabel` calls it.

File: docs/changes/latest.inc (hunk 3 above)
[the remark this answers is not in the document]
Comment (LeSasse) [reply docs/changes/latest.inc:66]: By copy, you mean move them over to the `test_merge_parcellations` tests only, such that no multiple parcellation scenario is tested in `test_parcel_aggregation` anymore?

File: junifer/data/parcellations.py (hunk 2 above)
Comment (LeSasse) [reply junifer/data/parcellations.py:714]: ok will try that, i think i put `niimg` before which the docs didn't recognise, but `niimg-like object`  makes sense

**Review by fraimondo — DISMISSED** [review]

**Review by synchon — DISMISSED** [review]

**Review by fraimondo — APPROVED** [review]

**Pull request merged**